pandora-server-directory/src/spaces/space.ts

```diff
@@ -349,7 +380,7 @@ export class Space {
 		// Ignore those - the requests will fail and once the space is not requestest for a bit, it will be unloaded from the directory too, actually vanishing
 	}
 
-	public checkAllowEnter(character: Character, data: { password?: string; invite?: SpaceInviteId; }, ignore: { characterLimit?: boolean; password?: boolean; } = {}): 'ok' | 'errFull' | 'noAccess' | 'invalidPassword' | 'invalidInvite' {
+	public checkAllowEnter(character: Character, data: { password?: string; invite?: SpaceInviteId; }, ignore: { characterLimit?: boolean; password?: boolean; } = {}): 'ok' | 'errFull' | 'noAccess' | 'invalidPassword' | 'invalidInvite' | 'spaceNotInUse' {
```

Comment:
Whole walking through this I noticed that an option to ignore password was added and that this option is used when querying _extended_ info about an existing room.
This is against the intention the code was written with - the _extended_ info should only be available if and only if the character can join a room - it is away to "peek" into the room without having to join (e.g. to see who is currently inside). I think people that don't have a password shouldn't be allowed to peek inside a password-protected room.

Reply:
ahh I see, honestly I thought this was a bug, and assumed if a room is public you should be able to see the extended info

Reply:
I changed this to `passwordOnInvite` this allows invites to work if they do not have `bypassPassword`

maybe invites could be changed to always bypass password